pkgs/build-support/rust/build-rust-crate/install-crate.nix

```diff
@@ -28,5 +29,23 @@ crateName: metadata:
       cp -P target/bin/* $out/bin # */
     fi
   fi
+  runHook postInstall
+'' else
+# for tests we just put them all in the output. No execution.
+''
+  runHook preInstall
+
+  mkdir -p $out/tests
+  if [ -e target/bin ]; then
+    find target/bin/ -type f -executable -exec cp {} $out/tests \;
+  fi
+  if [ -e target/lib ]; then
+    find target/lib/ -type f \! -name '*.rlib' \
+      -a \! -name '*${stdenv.hostPlatform.extensions.sharedLibrary}' \
+      -a \! -name '*.d' \
+      -executable \
+      -exec cp {} $out/tests \;
```

Comment:
Nitpick: This would create less processes for many files (not sure if that's a real world problem though)

```
find ... -0 | xargs -0 install --target $out/tests 
```

Reply:
Fixed